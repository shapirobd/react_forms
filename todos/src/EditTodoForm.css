.EditTodoForm {
	padding: 15px;
}
