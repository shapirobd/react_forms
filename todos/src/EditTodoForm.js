import React, { useState } from "react";
import "./EditTodoForm.css";

const EditTodoForm = ({
	editTodo,
	visibility,
	toggleEditFormVisibility,
	originalTodo,
}) => {
	const INITIAL_STATE = {
		task: "",
	};
	const [formData, setFormData] = useState(INITIAL_STATE);
	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value,
		}));
	};
	const handleSubmit = (e) => {
		e.preventDefault();
		editTodo(originalTodo, formData);
		setFormData(INITIAL_STATE);
		toggleEditFormVisibility();
	};
	return (
		<div className="EditTodoForm" style={{ display: visibility }}>
			<form onSubmit={handleSubmit}>
				<input
					type="text"
					id="task"
					name="task"
					placeholder="Edit todo"
					onChange={handleChange}
					value={formData.task}
				/>
				<button>Edit Todo</button>
			</form>
		</div>
	);
};

export default EditTodoForm;
