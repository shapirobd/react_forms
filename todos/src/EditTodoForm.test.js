import { render } from "@testing-library/react";
import EditTodoForm from "./EditTodoForm";

it("renders without crashing", () => {
	render(<EditTodoForm />);
});

it("matches snapshot", () => {
	const { asFragment } = render(<EditTodoForm />);
	expect(asFragment()).toMatchSnapshot();
});
