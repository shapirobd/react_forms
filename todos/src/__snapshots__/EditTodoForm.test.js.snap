// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="EditTodoForm"
  >
    <form>
      <input
        id="task"
        name="task"
        placeholder="Edit todo"
        type="text"
        value=""
      />
      <button>
        Edit Todo
      </button>
    </form>
  </div>
</DocumentFragment>
`;
